opentelemetry-sdk/tests/metrics/integration_test/test_console_exporter.py

```diff
@@ -20,18 +22,76 @@
     ConsoleMetricExporter,
     PeriodicExportingMetricReader,
 )
+from opentelemetry.test.globals_test import reset_metrics_globals
 
 
 class TestConsoleExporter(TestCase):
     def test_console_exporter(self):
 
-        try:
-            exporter = ConsoleMetricExporter()
-            reader = PeriodicExportingMetricReader(exporter)
-            provider = MeterProvider(metric_readers=[reader])
-            metrics.set_meter_provider(provider)
-            meter = metrics.get_meter(__name__)
-            counter = meter.create_counter("test")
-            counter.add(1)
-        except Exception as error:
-            self.fail(f"Unexpected exception {error} raised")
+        reset_metrics_globals()
+
+        output = StringIO()
+        exporter = ConsoleMetricExporter(out=output)
+        reader = PeriodicExportingMetricReader(
+            exporter, export_interval_millis=100
+        )
+        provider = MeterProvider(metric_readers=[reader])
+        metrics.set_meter_provider(provider)
+        meter = metrics.get_meter(__name__)
+        counter = meter.create_counter(
+            "name", description="description", unit="unit"
+        )
+        counter.add(1, attributes={"a": "b"})
+        provider.shutdown()
+
+        output.seek(0)
+        result_0 = loads(output.readlines()[0])
+
+        self.assertGreater(len(result_0), 0)
+        self.assertEqual(
```

Comment:
nit may be more readable to do a single `self.assertEqual(result_0, {"resource_metrics": [...]})`. I think python will give a nicer diff if something changes that way

Reply:
Can't actually compare against the entire output dictionary because there are some values (like timestamps) that can't be tested against. Refactored the code to remove repeated dictionary operations.